README.md

```diff
@@ -1,57 +1,253 @@
+  visit new_post_path
+
+  fill_in 'post_title', with: "My post title"
+  fill_in 'post_description', with: "My post description"
+
+  click_on "Submit Post"
+
+  expect(page).to have_content("My post title")
+end
+```
+
+This is a temporary spec since we eventually will want this to form to redirect to a newly created record page, but we'll use it to ensure our form is working properly. This fails for obvious reasons, let's follow the TDD process and let the failures help build our form. The first error says that it can't find the field ```post_title```. Let's add the following HTML form items into the ```_form.html.erb``` view template:
+
+```ERB
+<h3>Post Form</h3>
+
+<form>
+  <label>Post title:</label><br>
+  <input type="text" id="post_title" name="post_title"><br>
+
+  <label>Post Description</label><br>
+  <textarea id="post_description" name="post_description"></textarea><br>
+
+  <input type="submit" value="Submit Post">
+</form>
+```
+
+In looking at both of the input elements, I'm using the standard Rails convention:
+
+* ```id``` - This will have the model name followed by an underscore and then the attribute name
+
+* ```name``` - This is where Rails looks for the parameters and stores it in a Hash. In a traditional Rails application this will be nested inside of the model with the syntax ```model[attribute]```, however we will work through that in a later lesson.
+
+Ok, so the spec was able to fill in the form elements and submit, but it's giving an error because this form doesn't actually redirect to any page, let's first update the form so that it has an action:
+
+```ERB
+<form action="<%= posts_path %>">
+```
+
+This will now redirect to ```/posts```, however we also need to add a method so that the application knows that we are submitting form data via the POST HTTP verb:
+
+```ERB
+<form action="<%= posts_path %>" method="post">
+```
+
+Next we need to draw a route so that the routing system knows what to do when a POST request is sent to the ```/posts``` resource:
+
+```ruby
+resources :posts, only: [:index, :new, :create]
+```
+
+If you run rake routes, you will see we now have a ```posts#create``` action:
+
+```
+Prefix    Verb    URI                   Controller#Action
+posts     GET     /posts(.:format)      posts#index
+          POST    /posts(.:format)      posts#create
+new_post  GET     /posts/new(.:format)  posts#new
+post      GET     /post/:id(.:format)   posts#show
+```
+
+Now let's add in a ```create``` action in the posts' controller and have it grab the params, store them in an instance variable and then redirect to the new page:
+
+```ruby
+def create
+  session[:form_params] = params.inspect
+  redirect_to new_post_path
+end
+```
+
+I'm storing the values from the form in the session so when we redirect back to the form we can print these values out on the page to ensure the params are processing properly. Lastly, let's update the ```new.html.erb``` view template so it prints out the form values:
```

Comment:
huh this is an odd pattern. Why don't you just have it `puts` out? and then show the ouput. Or maybe tell them to put a binding.pry in there and then run `params.inspect`?